apps/antalmanac/src/actions/AppStoreActions.ts

```diff
@@ -86,10 +86,12 @@ export const saveSchedule = async (userID: string, rememberMe: boolean) => {
             try {
                 await trpc.users.saveUserData.mutate({ id: userID, userData: scheduleSaveState });
 
-                openSnackbar(
-                    'success',
-                    `Schedule saved under username "${userID}". Don't forget to sign up for classes on WebReg!`
-                );
+                if (!autoSave) {
+                    openSnackbar(
+                        'success',
+                        `Schedule saved under username "${userID}". Don't forget to sign up for classes on WebReg!`
+                    );
+                }
```

Comment:
This is an anti-pattern. Passing in booleans to mutate a function's behavior is like a half-baked builder pattern implementation. If you have specific functionality that you want to perform in a certain context, then take care of that in the same context instead of delegating it to another module that doesn't inherently understand it.

I'd be willing to go as far as just creating an entirely new function for auto saving, even if it means duplicating some code, just to keep these interfaces well defined. For example

```ts
export async function autosaveSchedule(userID: string, rememberMe: boolean) {
  // ...
}
```

Reply:
I added a specific function `autosaveSchedule` for auto save, and I remove the `rememberMe` in the function because autosave will not change the state of current user id in local storage.